include/ccf/base_endpoint_registry.h

```diff
@@ -82,9 +82,24 @@ namespace ccf
       const std::string& method_prefix_, ccfapp::AbstractNodeContext& context_);
 
     /** Get the status of a transaction by ID, provided as a view+seqno pair.
-     * This is a node-local property - while it will converge on all nodes in
-     * a healthy network, it is derived from distributed state rather than
-     * distributed itself.
+     *
+     * Note that this value is the node's local understanding of the status
+     * of that transaction in the network at call time. For a given TxID, the
+     * initial status is always UNKNOWN, and eventually becomes COMMITTED or
+     * INVALID. See the documentation section entitled "Verifying Transactions"
+     * for more detail.
+     *
+     *         UNKNOWN [Initial status]
+     *          v  ^
+     *        PENDING
+     *        v     v
+     *  COMMITTED INVALID [Final statuses]
```

Comment:
Are these transitions strictly true actually? For example, can't you jump from Unknown to Invalid directly? When would you transition back to Unknown once you're in Pending?

Reply:
1. Not all states are necessarily observed, depending on when you call. '`A -> B -> C` can look like `A->C` if you don't call when it was `B`. So while it's possible to go straight from `UNKNOWN` to a final state, I think it's acceptable not to try and display it here.
2. Yes you can, see above.
3. Node (primary) creates 3.5 [PENDING]. Rollback occurs on node to 2.4. 3.5 is now [UNKNOWN], until N.5 is applied where N>3 (received from new primary).